### PR TITLE
Outer loop support files

### PR DESCRIPTION
**Pull request by awisniew90**
Added outerloop Dockerfile and app-deploy.yaml

### CODE REVIEW

File: dev/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Step 1: Build the user's application
+FROM ajymau/java-openliberty-odo:0.1 as compile
```
Comment (scottkurz) [inline dev/Dockerfile:2]: Should this move to 0.1.1 now?

File: dev/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# test to see if each is present and move to a well known location for later processing in the next stage
+# 
+RUN cd /project/user-app/target/liberty/wlp/usr/servers && \
+    if [ -d ./defaultServer/lib ]; then mv ./defaultServer/lib /configlibdir; fi && \
+    if [ ! -d /configlibdir/lib ]; then mkdir /configlibdir/lib; fi && \
+    mv -f defaultServer/* /config/ && \
+    if [ -d ../shared ]; then mv ../shared/* /shared/; fi
+
+# Step 2: Package Open Liberty image
+FROM openliberty/open-liberty:kernel-java8-openj9-ubi
```
Comment (scottkurz) [inline dev/Dockerfile:32]: Should match our devfile version: OL 20.0.0.6

File: templates/default/src/main/liberty/config/configDropins/defaults/quick-start-security.xml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<server>
+    <quickStartSecurity userName="admin" userPassword="adminpwd" />
```
Comment (scottkurz) [inline templates/default/src/main/liberty/config/configDropins/defaults/quick-start-security.xml:2]: Should we use the generated keystore password?

File: dev/Dockerfile (hunk 2 above)
Comment (scottkurz) [reply dev/Dockerfile:32]: actually I think Andy's changing to 20.0.0.9

File: dev/Dockerfile (hunk 1 above)
Comment (awisniew90) [reply dev/Dockerfile]: updated to 0.1.3 (the latest now)

File: dev/Dockerfile (hunk 2 above)
Comment (awisniew90) [reply dev/Dockerfile]: Updated to 20.0.0.9

File: templates/default/src/main/liberty/config/configDropins/defaults/quick-start-security.xml (hunk 3 above)
Comment (awisniew90) [reply templates/default/src/main/liberty/config/configDropins/defaults/quick-start-security.xml:2]: Updated this issue to track this change: https://github.com/OpenLiberty/application-stack/issues/21

**Comment by awisniew90** [comment]
I opened this issue to track adding outerloop support for java 8: https://github.com/OpenLiberty/application-stack/issues/49. This might happen by default since our java 8 devfile is an extension, but we still need to make sure the outerloop and inner loop are using the same java version. right now our outerloop is just using Java 11. 

https://github.com/OpenLiberty/application-stack/issues/49

**Review by scottkurz — APPROVED** [review]
LGTM

**Comment by awisniew90** [comment]
workaround for the time being (according to what's in https://github.com/openshift/odo/pull/3478) is to create a url prior to `odo deploy` to allow the PORT variable to populate correctly in our app-deploy.yaml. For example:

1. odo url create --port=9080
2. odo deploy --tag image-registry.openshift-image-registry.svc:5000/<namespace>/<project name>

**Pull request merged**